### DISCUSSION TITLE
Add `LARAVEL_START`

### DISCUSSION THREAD

**Discussion by bilogic** [orchestral/testbench#387] [Q&A] [answered]
Hi!

My commands rely on `LARAVEL_START` so that I can constantly keep an eye out for performance issues. Can also define it so that my code don't have to work around it? I can PR.

Thank you!

**Comment by crynobone** [comment]
what 's the use case?

**Comment by bilogic** [comment]
1. My custom artisan commands always show the amount of time taken from `LARAVEL_START` until exit, so that all developers are exposed to this simple performance metric constantly.

2. One of the command is a overridden version of `migrate:fresh` which is executed during our CI/CD and because it cannot find `LARAVEL_START`, the test fails.

3. I suppose `define('LARAVEL_START', microtime(true));` at the start of testbench's first PHP file would have minimal impact, right?

**Comment by crynobone** [comment]
Feel free to submit the PR to https://github.com/orchestral/testbench-core/blob/0b7de82296358c3013fe0a3d1128cbb6dc33f801/src/Console/Commander.php#L97 and target branch `6.x`

**Comment by bilogic** [comment]
I added `define('LARAVEL_START', microtime(true));` at your location, but it did not solve my problem.

Eventually, I fixed it by adding `define('LARAVEL_START', microtime(true));` at these locations

## For Laravel projects
https://github.com/laravel/framework/blob/c6aeffded35c10ac60b14d4cff4b6c45dc57e9ea/src/Illuminate/Foundation/Testing/TestCase.php#L85

## For Laravel packages
The equivalent in testbench-core would be https://github.com/orchestral/testbench-core/blob/158e2f483bccab28f36d6170a551641534813886/src/TestCase.php#L48

Or is there a better place to define `LARAVEL_START`?

**Comment by crynobone** [comment]
In your `phpunit.xml`:

```diff
-bootstrap="vendor/autoload.php"
+bootstrap="tests/bootstrap.php"
```

And then add `tests/bootstrap.php` with following content:

```php
<?php

require __DIR__.'/../vendor/autoload.php';

define('LARAVEL_START', microtime(true));
```

**Reply by bilogic** [reply]
Thanks but it also means I need to change close to a hundred `phpunit.xml`

**Reply by crynobone** [reply, marked as answer]
No plan to add constant under TestCase

**Reply by hayleyberryl2s** [reply]
Laravel (`./public/index.php`, `./artisan`) puts the define before `require`-ing the `autoload.php`. The autoload.php runs code and loads a bunch of dependencies from the filesystem, so putting `define('LARAVEL_START' ...` before that will provide a more realistic execution start time.
